src/docs/arch-fast-truncate.dox

```diff
@@ -0,0 +1,847 @@
+reconciliation result exists it can be used to reconcile the parent.
+(This is true whether or not it includes the tombstones from instantiation, i.e., whether
+the truncation was committed.)
+
+Note that "unresolved" includes "prepared".
+While we cannot write out a prepared truncation in the parent page's address cells, in
+principle after instantiation we could write out the prepared tombstone updates like any
+other prepared updates, and after doing so nothing special is needed in the parent.
+This is not currently done, chiefly because safely determining whether the truncation is
+prepared at the point where eviction needs to check is problematic.
+(We can't check \c page_del because it might have been discarded by then; looking in \c
+inst_updates is at best messy and it isn't entirely clear what locking or synchronization
+might be needed.
+Some such check, or an extra flag in the modify structure, is probably possible if it
+eventually becomes important to allow these evictions.)
+
+@section ft_on_disk_format On-disk format
+
+Deleted pages are referred to on disk by a special address cell type, \c WT_CELL_ADDR_DEL.
+These contain three additional packed integers between the time aggregate and the cell
+data: the transaction ID, commit timestamp, and durable timestamp of the transaction that
+deleted the page.
+(Only committed truncations are written out.
+Prepared truncations cannot be represented on disk.
+Truncations that are globally visible do not result in a cell in the parent page at all.)
+
+These fields are, however, only present if the page header includes the \c
+WT_PAGE_FT_UPDATE flag, whose value is 0x20.
+Proper support for timestamped fast-truncate only appeared in WT 11.0; earlier versions
+neither write these fields nor expect to find them.
+The explicit header flag is required to make compatibility guarantees function as needed.
+(MongoDB 6.x knows how to read pages with \c WT_PAGE_FT_UPDATE set, but does not write
+them.
+This is controlled in the WT library by the build config.)
+
+@section ft_truncate Truncation
+
+The B-tree cursor truncate code, found in \c bt_cursor.c, iterates through the specified
+truncation range with \c cursor_next, individually removing all values it finds.
+This is the slow-truncate path.
+
+The fast-truncate functionality is implicit.
+Passing \c true for the \c truncating argument of \c __wt_btcur_next causes the flag \c
+WT_READ_TRUNCATE to be passed to the tree-walk inside the next code.
+Fast-truncate then happens inside the tree-walk code (in \c bt_walk.c) which calls \c
+__wt_delete_page on leaf pages it visits.
+
+This function, in \c bt_delete.c, checks the page for eligibility.
+First, if the page is unmodified and in memory we attempt to evict it.
+Then we check if the page is on disk, that is, the state is \c WT_REF_DISK, and if so lock
+the ref.
+(And then check if it is <i>still</i> on disk.)
+
+Then the following categories of pages are ineligible for fast-truncate:
```

Comment:
Is it possible to have bullet points or a simple list for the different reasons listed in this paragraph?

Reply:
It is supposed to be! The doxygen manual says that's how to make a bullet list. Apparently it's just lying and you need to use - and not *. Fixed in [af26591](https://github.com/wiredtiger/wiredtiger/pull/8237/commits/af265919038ffe7086fbdc1db36c6800c63bc466). Also, I fixed up the wording of the overflow items entry.

Reply:
The bullet point list is working now! And thanks for the rewording.